sdks/java/io/tika/src/main/java/org/apache/beam/sdk/io/tika/TikaIO.java

```diff
@@ -0,0 +1,307 @@
+     *
+     * <p>Filepattern can be a local path (if running locally), or a Google Cloud Storage
+     * filename or filename pattern of the form {@code "gs://<bucket>/<filepath>"}
+     * (if running locally or using remote execution service).
+     *
+     * <p>Standard <a href="http://docs.oracle.com/javase/tutorial/essential/io/find.html" >Java
+     * Filesystem glob patterns</a> ("*", "?", "[..]") are supported.
+     */
+    public Read from(String filepattern) {
+      checkNotNull(filepattern, "Filepattern cannot be empty.");
+      return from(StaticValueProvider.of(filepattern));
+    }
+
+    /** Same as {@code from(filepattern)}, but accepting a {@link ValueProvider}. */
+    public Read from(ValueProvider<String> filepattern) {
+      checkNotNull(filepattern, "Filepattern cannot be empty.");
+      return toBuilder()
+          .setFilepattern(filepattern)
+          .setQueuePollTime(Read.DEFAULT_QUEUE_POLL_TIME)
+          .setQueueMaxPollTime(Read.DEFAULT_QUEUE_MAX_POLL_TIME)
+          .build();
+    }
+
+    /**
+     * Returns a new transform which will use the custom TikaConfig.
+     */
+    public Read withTikaConfigPath(String tikaConfigPath) {
+      checkNotNull(tikaConfigPath, "TikaConfigPath cannot be empty.");
+      return withTikaConfigPath(StaticValueProvider.of(tikaConfigPath));
+    }
+
+    /** Same as {@code with(tikaConfigPath)}, but accepting a {@link ValueProvider}. */
+    public Read withTikaConfigPath(ValueProvider<String> tikaConfigPath) {
+      checkNotNull(tikaConfigPath, "TikaConfigPath cannot be empty.");
+      return toBuilder()
+          .setTikaConfigPath(tikaConfigPath)
+          .build();
+    }
+
+    /**
+     * Returns a new transform which will use the provided content type hint
+     * to make the file parser detection more efficient.
+     */
+    public Read withContentTypeHint(String contentType) {
+      checkNotNull(contentType, "ContentType cannot be empty.");
+      Metadata metadata = new Metadata();
+      metadata.add(Metadata.CONTENT_TYPE, contentType);
+      return withInputMetadata(metadata);
+    }
+
+    /**
+     * Returns a new transform which will use the provided input metadata
+     * for parsing the files.
+     */
+    public Read withInputMetadata(Metadata metadata) {
+      Metadata inputMetadata = this.getInputMetadata();
+      if (inputMetadata != null) {
+        for (String name : metadata.names()) {
+            inputMetadata.set(name, metadata.get(name));
+        }
+      } else {
+        inputMetadata = metadata;
+      }
+      return toBuilder().setInputMetadata(inputMetadata).build();
+    }
+
+    /**
+     * Returns a new transform which will report the metadata.
+     */
+    public Read withReadOutputMetadata(Boolean value) {
+      return toBuilder().setReadOutputMetadata(value).build();
+    }
+
+    /**
+     * Returns a new transform which will use the specified queue poll time.
```

Comment:
Generally, "Returns a new transform which ..." is redundant - other IOs don't say this in the doc, it is implied because it is true for all transforms in Beam.